plugins/enthusiast-common/enthusiast_common/agents/base.py

```diff
@@ -0,0 +1,35 @@
+from abc import ABC, abstractmethod
+
+from langchain.agents import AgentExecutor
+from langchain.memory import ConversationSummaryBufferMemory
+from langchain_core.language_models import BaseLanguageModel
+from langchain_core.prompts import ChatPromptTemplate
+from langchain_core.tools import BaseTool
+
+from ..services import BaseConversationService
+
+
+class BaseAgent(ABC):
+    _agent_executor: AgentExecutor
+
+    def __init__(
+        self,
+        tools: list[BaseTool],
+        llm: BaseLanguageModel,
+        prompt: ChatPromptTemplate,
+        conversation_service: BaseConversationService,
+        conversation_id: int,
+        agent_executor: AgentExecutor,
```

Comment:
To be reviewed: should this be a part of Agent's interface, or is that something that Agent's implementation can provision for itself?